### DISCUSSION TITLE
Default pipe parameter

### DISCUSSION THREAD

**Discussion by DSchroer** [DSchroer/dslcad#18] [General]
> As an aside, how would you feel making shape the default argument for ->, or perhaps the first declared parameter for any function/file.

This is a good idea to look into. I feel that there should be a better solution that we can come to before implementing something. I have concerns with both of these options right now. 

For using `shape` by default:
1. If we use shape that limits the usage and forces shape to be a default

For using the first parameter:
1. Parameters don't really have order in them so the first parameter doesn't really fit. (Maybe they should have order)
2. Overloaded functions could break existing code. Take `sin(radians)` for an example. If you have `5 -> sin()` and then I add `sin(degrees)` which one should the pipe call? `

**Comment by ephetic** [comment]
For really obvious cases, you could introduce a `default` keyword.  In a file it would be like
```
// -> ./center()
default var shape;
var x = true;
var y = true;
var z = true;
// ...
```

This would work the same way with the local scope/function syntax I suggested in #13, with a related question I'll add there.

I'm not sure how you would add it to functions defined in Rust, nor if you plan to allow user defined functions.
